travis-script.sh

```diff
@@ -1,8 +1,39 @@
 #!/usr/bin/env bash
 set -ev
 
+# ---------------------------------------------------------------------
+# Bootstrap cabal, to verify bootstrap.sh script works.
+# ---------------------------------------------------------------------
+
+OLD_CWD=$PWD
+
+# Bootstrap
+cd Cabal
+ghc --make Setup
+./Setup configure --user
+./Setup build
+./Setup install
+cd ../cabal-install
+env EXTRA_CONFIGURE_OPTS="" ./bootstrap.sh --no-doc
```

Comment:
So the problem now is that when we install Cabal before bootstrapping, Cabal will be compiled against the version of binary that came with GHC. I think we should modify `bootstrap.sh` to support installing the currently checked-out version of Cabal, then we can install it after `binary`.